framework/src/auxkernels/GhostingAux.C

```diff
@@ -0,0 +1,83 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GhostingAux.h"
+
+registerMooseObject("MooseApp", GhostingAux);
+
+template <>
+InputParameters
+validParams<GhostingAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+
+  params.addClassDescription("Colors the elements ghosted to the chosen PID.");
+
+  params.addRequiredParam<processor_id_type>("pid", "The PID to see the ghosting for");
+
+  MooseEnum functor_type("geometric algebraic", "geometric");
+
+  params.addParam<MooseEnum>("functor_type", functor_type, "The type of ghosting functor to use");
+
+  return params;
+}
+
+GhostingAux::GhostingAux(const InputParameters & parameters)
+  : AuxKernel(parameters),
+    _pid(getParam<processor_id_type>("pid")),
+    _functor_type(getParam<MooseEnum>("functor_type"))
+{
+  if (isNodal())
+    mooseError("GhostingAux only works on elemental fields.");
+
+  // Can only work on Replicated meshes
+  _mesh.errorIfDistributedMesh("GhostingAux");
```

Comment:
This _really_ should be made to work with DistributedMesh. Figuring out the ghosting on ReplicatedMesh feels more like a testing exercise. Is this a MOOSE utility or not? This object is adding a new API to AuxKernel so it really should be first class IMHO.